examples/shiftbot/src/map.ts

```diff
@@ -0,0 +1,227 @@
+
+import Client, { Animation, Block, Structure } from '../../../dist/index.js'
+import client from './shift.js'
+import fs, { writeFileSync } from 'node:fs'
+import path from 'node:path'
+
+import { is_bot_admin } from './admin.js'
+
+export const width = 50
+export const height = 37
+
+const TOP_LEFT = { x: parseInt(process.env.TLX || '0'), y: parseInt(process.env.TLY || '0') }
+const QUEUE: [string, string][] = []
+const RECENT_MAPS: string[] = []
+const MAPS_PATH = process.env.MAPS_PATH || 'maps'
+
+const map = new Structure(50, 37)
+const map_without_doors = new Structure(50, 37)
+
+function get_map(): Structure {
+    let maps = fs.readdirSync(MAPS_PATH).filter(p => !p.startsWith('filled') && !p.startsWith('empty')),
+        map_path: string
+
+    if (QUEUE.length > 0)
+        map_path = QUEUE.shift()?.[0] as string
+    else {
+        maps = maps.filter(p => !RECENT_MAPS.includes(p))
+        map_path = maps[Math.floor(maps.length * Math.random())]
+    }
+
+    RECENT_MAPS.push(map_path)
+    if (RECENT_MAPS.length >= maps.length / 2) RECENT_MAPS.shift()
+
+    return Structure.fromString(
+        fs.readFileSync(path.join(MAPS_PATH, map_path))
+            .toString('ascii'))
+}
+
+function encode_doors() {
+    // Bottom Door
+    for (let x = 1; x < map.width - 1; x++) {
+        if (map_without_doors.foreground[x][map.height - 3].name == 'key_door_green') {
+            map_without_doors.foreground[x][map.height - 5] = new Block('gravity_up')
```

Comment:
`for` loop? Here and in the others below

Reply:
I have unnested the if, unsure if a for loop would make sense for 3 entries in https://github.com/Anatoly03/pixelwalker.js/pull/18/commits/df4b6cc60b0f2dda2e5c389f5dbf2fd52d081c6f